### PR TITLE
Fix printing of XML on parse failure

### PR DESCRIPTION
**Pull request by Beyley**

### CODE REVIEW

File: Bunkum.Core/Endpoints/Middlewares/MainMiddleware.cs [hunk 1]
```diff
@@ -292,7 +292,7 @@ private bool TryAddBodyToInvocation(EndpointAttribute attribute, Type paramType,
             }
             catch (Exception e)
             {
-                this._logger.LogError(BunkumCategory.UserContent, $"Failed to parse object data: {e}\n\nXML: {body}");
+                this._logger.LogError(BunkumCategory.UserContent, $"Failed to parse object data: {e}\n\nXML: {Encoding.UTF8.GetString(body.ToArray())}");
```
Comment (jvyden) [inline Bunkum.Core/Endpoints/Middlewares/MainMiddleware.cs]: This won't always be a MemoryStream in the future (see #67), so I'd prefer if we read everything out once at the top of the function and then used that for everything.

Might as well refactor this function while we're here

**Review by jvyden — CHANGES_REQUESTED** [review]
Have you checked if this also happens with JSON?

**Review by jvyden — APPROVED** [review]

**Comment by jvyden** [comment]
Actually, before I merge, do we know if this affects performance or fixes the null byte bug?

**Comment by Beyley** [comment]
> Actually, before I merge, do we know if this affects performance or fixes the null byte bug?

Performance should be exactly the same (maybe slightly faster?), and it does not fix the null byte issue, the request itself going over the wire has the null bytes, and its not just a single null byte, the game itself is reading out of bounds when sending the data

**Pull request merged**